src/App.tsx

```diff
@@ -15,47 +16,118 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+const getReorderedGoods = (
+  goods: string[],
+  sortBy: SortType,
+  reversed: boolean,
+) => {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((a, b) => {
+    switch (sortBy) {
+      case SortType.ALPHABET:
+        return a.localeCompare(b);
+
+      case SortType.LENGTH:
+        return a.length - b.length;
+
+      default:
+        return 0;
+    }
+  });
+  if (reversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+};
+
+export const App: FC = () => {
+  const [isReversed, setReverse] = useState(false);
```

Comment:
Let's follow the naming convention for the useState. 
`const [value, setValue] = useState()`
```suggestion
  const [isReversed, setIsReversed] = useState(false);
```